### PR TITLE
Add num_constraints for counting all constraints in a model

### PR DESCRIPTION
**Pull request by odow**
As requested by @ccoffrin on Gitter.

### CODE REVIEW

File: src/constraints.jl [hunk 1]
```diff
@@ -1203,3 +1203,45 @@ function list_of_constraint_types(model::Model)::Vector{Tuple{Type,Type}}
...
+        model::Model;
+        count_variable_in_set_constraints::Bool = true,
+    )
+
+Return the number of constraints in `model`.
+
+If `count_variable_in_set_constraints`, then constraints such as
+`VariableRef`-in-`Integer` are included. To count only the number of structural
+constraints (e.g., the rows in the constraint matrix of a linear program), pass
+`count_variable_in_set_constraints = false`.
```
Comment (blegat) [inline src/constraints.jl:1218]: It's a bit ambiguous as to whether `VectorOfVariables` constraints are counted.

File: src/constraints.jl [hunk 2]
```diff
@@ -1203,3 +1203,45 @@ function list_of_constraint_types(model::Model)::Vector{Tuple{Type,Type}}
...
+julia> num_constraints(model)
+3
+
+julia> num_constraints(model; count_variable_in_set_constraints = false)
+1
+```
+"""
+function num_constraints(
+    model::Model;
+    count_variable_in_set_constraints::Bool = true,
```
Comment (mlubin) [inline src/constraints.jl:1238]: IMO this keyword argument should be required, no default value.

Comment (odow) [reply src/constraints.jl]: Done

**Review by blegat — APPROVED** [review]

File: src/constraints.jl (hunk 1 above)
Comment (blegat) [reply src/constraints.jl]: Yes the phrasing was a bit ambiguous, it's much clearer now

File: src/constraints.jl [hunk 3]
```diff
@@ -1203,3 +1203,39 @@ function list_of_constraint_types(model::Model)::Vector{Tuple{Type,Type}}
         (F, S) in MOI.get(model, MOI.ListOfConstraintTypesPresent())
     ]
 end
+
+"""
+    num_constraints(model::Model; count_variable_in_set_constraints::Bool)
+
+Return the number of constraints in `model`.
+
+If `count_variable_in_set_constraints = true`, then `VariableRef` constraints
```
Comment (mlubin) [inline src/constraints.jl:1212]: Nit: ``== true``

**Review by mlubin — APPROVED** [review]

File: src/constraints.jl (hunk 3 above)
Comment (odow) [inline src/constraints.jl:1212]:
```suggestion
If `count_variable_in_set_constraints == true`, then `VariableRef` constraints
```

**Pull request merged**